src/transformers/generation_logits_process.py

```diff
@@ -167,6 +167,32 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         return scores
 
 
+class HallucinationPenaltyLogitsProcessor(LogitsProcessor):
+    r"""
+    [`LogitsProcessor`] enforcing an exponential penalty on sequences that are not in the original input.
+
+    Args:
+        hallucination_penalty (`float`):
+            The parameter for hallucination penalty. 1.0 means no penalty.
+    """
+
+    def __init__(self, penalty: float, encoder_input_ids: torch.LongTensor):
+        if not isinstance(penalty, float) or not (penalty > 0):
+            raise ValueError(f"`penalty` has to be a strictly positive float, but is {penalty}")
+
+        self.penalty = penalty
+        self.encoder_input_ids = encoder_input_ids
+
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
+        score = torch.gather(scores, 1, self.encoder_input_ids)
+
+        # if score < 0 then repetition penalty has to be multiplied to reduce the previous token probability
+        score = torch.where(score < 0, score / self.penalty, score * self.penalty)
```

Comment:
```suggestion
        score = torch.where(score < 0, score * self.penalty, score / self.penalty)
```
-> can we write it this way and make it a "penalty" logits processor so that it's 1-to-1 aligned with the one we already have? 
Then we can just pass `1 / penalty` instead of `penalty` for your use case :-)

Reply:
Sure thing, I just changed the penalty and the name to `EncoderRepetitionPenaltyLogitsProcessor`. Let me know how that looks now :)